### ISSUE TITLE
PyTorch ONNX export is exporting torch.cat layer as onnx::Constant[value=<Tensor>]

### ISSUE THREAD

**Issue by khurramHazen** [pytorch/pytorch#12220]
## ❓ Questions and Help
I am trying to export my own implemented YOLOv3 from PyTorch to ONNX (To be used in Caffe2+TensorRT). The outputs of the network are like this:

```
y1 = self.yolo_82(x)
...
y2 = self.yolo_94(x)
...
y3 = self.yolo_106(x)
return torch.cat( (y1,y2,y3), 1 )
```

The issue is, the ONNX exporter is exporting this last layer as: 
```
{
  %439 : Float(1, 10647, 85) = onnx::Constant[value=<Tensor>]()
  return (%439);
}

```
As a result when I run the model in Caffe2, it gives me the same output as on the image I used to export the network. 
My questions are:
1. [PyTorch Docs](https://pytorch.org/docs/0.3.1/onnx.html) claim that torch.cat is supported in ONNX export, if yes then why is the ONNX export returning a constant array instead of concatenated tensors?
2. Is there some special/secret ingredient that I am missing here.

Any/all help is appreciated as I am stuck here for quite long.

### Please note that this issue tracker is not a help form and this issue will be closed.

We have a set of [listed resources available on the website](https://pytorch.org/resources). Our primary means of support is our discussion forum:

- [Discussion Forum](https://discuss.pytorch.org/)

**Comment by khurramHazen** [comment]
Sorry my bad I was using `.detach()` and `.data` during forward pass (as the code was written for training in PyTorch and it is my first export to ONNX). Hence the ONNX was (rightly) complaining about constants as for it the tensors were constants and not PyTorch's graph nodes.

**Comment by liangjing1** [comment]
Hello, I also encountered the same problem. Can you tell me how did you solve it？
`a = torch.ones((1, 1, 100, 60)).float()
b = torch.zeros((1, 1, 100, 60)).float()
mat0 = torch.cat((a, b), 1)
mat1 = torch.cat((b, a), 1)`

 %80 : Float(1, 2, 100, 60) = onnx::Constant[value=<Tensor>]()
